jupyterhub/handlers/pages.py

```diff
@@ -308,10 +308,13 @@ async def _wrap_spawn_single_user(
         # otherwise it may cause a redirect loop
         if f.done() and f.exception():
             exc = f.exception()
+            self.log.exception(f"Error starting server {spawner._log_name}: {exc}")
+            if isinstance(exc, web.HTTPError):
+                # allow custom HTTPErrors to pass through
+                raise
```

Comment:
It took me a while to understand this Python syntax (https://stackoverflow.com/questions/25632147/raise-at-the-end-of-a-python-function-outside-try-or-except-block)

Is it worth being explicit and using `raise exc`, or does that behave differently?

Reply:
Good catch, this is definitely wrong!

A naked `raise` is only appropriate inside an `except:`  block (`raise` is basically "cancel the `except` block," whereas `raise exc` modifies the traceback). It should be `raise exc` here.